internal/pkg/action/operator_uninstall.go

```diff
@@ -57,16 +65,41 @@ func (u *OperatorUninstall) Run(ctx context.Context) error {
 		return fmt.Errorf("operator package %q not found", u.Package)
 	}
 
-	csv, err := u.getInstalledCSV(ctx, sub)
-	if err != nil {
-		return fmt.Errorf("get installed CSV %q: %v", sub.Status.InstalledCSV, err)
+	var subObj, csvObj controllerutil.Object
+	var crds []controllerutil.Object
+	if sub != nil {
+		subObj = sub
+		// CSV name may either be the installed or current name in a subscription's status,
+		// depending on installation state.
+		csvKey := types.NamespacedName{
+			Name:      sub.Status.InstalledCSV,
+			Namespace: u.config.Namespace,
+		}
+		if csvKey.Name == "" {
+			csvKey.Name = sub.Status.CurrentCSV
+		}
+
+		// This value can be empty which will cause errors.
+		if csvKey.Name != "" {
+			csv := &v1alpha1.ClusterServiceVersion{}
+			if err := u.config.Client.Get(ctx, csvKey, csv); err != nil && !apierrors.IsNotFound(err) {
+				return fmt.Errorf("error getting installed CSV %q: %v", csvKey.Name, err)
+			} else if err == nil {
+				crds = getCRDs(csv)
+			}
+			csvObj = csv
```

Comment:
(optional) you could move all of this into `OperatorUninstall.getInstalledCSV()`.